.github/workflows/reusable-create-collector-contrib-pull-request.yml

```diff
@@ -0,0 +1,75 @@
+name: Reusable - Create collector contrib pull request
+
+on:
+  workflow_call:
+    inputs:
+      jmx-metrics-version:
+        type: string
+        required: true
+      jmx-metrics-sha256:
+        type: string
+        required: true
+    secrets:
+      BOT_TOKEN:
+        required: true
+  # to help with partial release build failures
+  workflow_dispatch:
+    inputs:
+      jmx-metrics-version:
+        description: "JMX metrics version"
+        required: true
+      jmx-metrics-sha256:
+        description: "JMX metrics hash (sha256)"
+        required: true
+
+jobs:
+  create-collector-contrib-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: opentelemetry-java-bot/opentelemetry-collector-contrib
+          # this is the personal access token used for "git push" below
+          token: ${{ secrets.JAVA_BOT_TOKEN }}
+
+      - name: Initialize pull request branch
+        env:
+          VERSION: ${{ inputs.jmx-metrics-version }}
+        run: |
+          git remote add upstream https://github.com/open-telemetry/opentelemetry-collector-contrib.git
+          git fetch upstream
+          git checkout -b update-opentelemetry-jmx-metrics-to-${VERSION} upstream/main
+
+      - name: Update version
+        env:
+          VERSION: ${{ inputs.jmx-metrics-version }}
+          HASH: ${{ inputs.jmx-metrics-sha256 }}
+        run: |
+          # NOTE there are intentional tab characters in the line below
+          sed "/^var jmxMetricsGathererVersions/a \	\"$HASH\": {\n		version: \"$VERSION\",\n		jar:     \"JMX metrics gatherer\",\n	}," receiver/jmxreceiver/supported_jars.go
```

Comment:
Nit: should we add a comment in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/jmxreceiver/supported_jars.go saying that this variable is auto-updated from here and should not be changed/renamed?

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/11456